realsense/plugin.cpp

```diff
@@ -36,17 +43,65 @@ class realsense : public plugin {
+        }
+    void auto_select_cam(){
+        rs2::context ctx;
+        rs2::device_list devices = ctx.query_devices();
+        for (rs2::device device : devices) {
+            std::string name = "Unknown Device";
+            if (device.supports(RS2_CAMERA_INFO_NAME)) {
+                name = device.get_info(RS2_CAMERA_INFO_NAME);
+                if (name.find("D4") != std::string::npos){
+                    std::cout << "Found D4XX Device, Checking for supported sensors" << std::endl;
+                    std::vector<rs2::sensor> sensors = device.query_sensors();
+                    for (rs2::sensor sensor : sensors){
+                        if (sensor.supports(RS2_CAMERA_INFO_NAME)){
+                            std::string sensor_name = sensor.get_info(RS2_CAMERA_INFO_NAME);
+                            if (sensor_name.find("Motion Module") != std::string::npos){
+                                std::cout << "Setting realsense_cam: D4XX" << std::endl;
+                                realsense_cam = "D4XX";
+                                break;
+                            }
+                        }
+                    }
+                    if (realsense_cam.compare("D4XX") == 0){
+                        break;
+                    }
+                } 
+                else if (name.find("T26") != std::string::npos){
+                    realsense_cam = "T26X";
+                    std::cout << "Setting realsense_cam: T26X" << std::endl;
+                }
+            }
+        }
+    }
+    void configure_camera()
+    {
+        if (realsense_cam.compare("auto") == 0) {
+            auto_select_cam();
+        }
+        if (realsense_cam.compare("T26X") == 0){
+            cfg.enable_stream(RS2_STREAM_ACCEL, RS2_FORMAT_MOTION_XYZ32F); // adjustable to 0, 63 (default), 250 hz
+            cfg.enable_stream(RS2_STREAM_GYRO, RS2_FORMAT_MOTION_XYZ32F); // adjustable set to 0, 200 (default), 400 hz
+            cfg.enable_stream(RS2_STREAM_FISHEYE, 1, RS2_FORMAT_Y8);
+            cfg.enable_stream(RS2_STREAM_FISHEYE, 2, RS2_FORMAT_Y8);
+            profiles = pipe.start(cfg, [&](const rs2::frame& frame) { this->callback(frame); });
+        }
+        else if (realsense_cam.compare("D4XX") == 0){
+            cfg.enable_stream(RS2_STREAM_ACCEL, RS2_FORMAT_MOTION_XYZ32F, ACCEL_RATE_D4XX); // adjustable to 0, 63 (default), 250 hz
+            cfg.enable_stream(RS2_STREAM_GYRO, RS2_FORMAT_MOTION_XYZ32F, GYRO_RATE_D4XX); // adjustable set to 0, 200 (default), 400 hz
+            cfg.enable_stream(RS2_STREAM_INFRARED, 1, IMAGE_WIDTH_D4XX, IMAGE_HEIGHT_D4XX, RS2_FORMAT_Y8, FPS_D4XX);
+            cfg.enable_stream(RS2_STREAM_INFRARED, 2, IMAGE_WIDTH_D4XX, IMAGE_HEIGHT_D4XX, RS2_FORMAT_Y8, FPS_D4XX);
+            cfg.enable_stream(RS2_STREAM_COLOR, IMAGE_WIDTH_D4XX, IMAGE_HEIGHT_D4XX, RS2_FORMAT_BGR8, FPS_D4XX);
+            cfg.enable_stream(RS2_STREAM_DEPTH, IMAGE_WIDTH_D4XX, IMAGE_HEIGHT_D4XX, RS2_FORMAT_Z16, FPS_D4XX);
```

Comment:
Also, if it helps to simplify the logic, we can introduce a generic intermediate value for shared options:
```cpp
switch_case (realsense_cam) {
    case RS_CAM_D4XX: 
        image_width = IMAGE_WIDTH_DX44;
        break;
    case RS_CAM_T26X:
        image_width = IMAGE_WIDTH_T26X:
        break;
    default:
        throw std::runtime_error("Unexpected camera value for 'realsense_cam'");
}
```
If there area a lot of shared camera properties, we can use a properties `struct` to organize them.

Reply:
I switched to detecting supported hardware using the streams they provide. It would be possible to search for an array of requested streams or data types if that would be useful in the future.